components/dashboard/Switcher.tsx

```diff
@@ -0,0 +1,258 @@
+  const childrenAdminMemberships = flatten(adminMemberships?.map(m => m.collective.children)).map(c => ({
+    collective: c,
+    role: 'ADMIN', // Admin role is inherited from parent
+  }));
+  const uniqueMemberships = uniqBy([...(adminMemberships ?? []), ...childrenAdminMemberships], m => m.collective.id);
+
+  return uniqueMemberships.filter(Boolean);
+});
+
+const getAdminMemberships = memoizeOne(loggedInUser => {
+  // Personal profile already includes incognito contributions
+  const adminMemberships = loggedInUser?.memberOf?.filter(
+    m => m.role === 'ADMIN' && !m.collective.isIncognito && !m.collective.isArchived,
+  );
+
+  // Filter out something if I am also an admin of the parent
+  const childrenAdminAccountIds = flatten(adminMemberships?.map(m => m.collective.children)).map((c: any) => c.id);
+
+  return adminMemberships?.filter(m => !childrenAdminAccountIds.includes(m.collective.id));
+});
+
+const getAdminMembershipsForArchived = memoizeOne(loggedInUser => {
+  // Personal profile already includes incognito contributions
+  const adminMembershipsForArchived = loggedInUser?.memberOf?.filter(
+    m => m.role === 'ADMIN' && !m.collective.isIncognito && m.collective.isArchived,
+  );
+  return adminMembershipsForArchived;
+});
+
+const MenuEntry = ({ account, activeSlug }) => {
+  const [expanded, setExpanded] = React.useState(false);
+  return (
+    <React.Fragment>
+      <StyledMenuEntry
+        key={account.id}
+        href={`/dashboard/${account.slug}`}
+        title={account.name}
+        $isActive={activeSlug === account.sug}
+      >
+        <Flex overflow="hidden" alignItems="center" gridGap="12px">
+          <Avatar collective={account} size={32} />
+          <Span truncateOverflow>{account.name}</Span>
+        </Flex>
+        {account.children?.length > 0 && (
+          <StyledRoundButton
+            ml={2}
+            size={24}
+            color="#C4C7CC"
+            onClick={e => {
+              e.preventDefault();
+              e.stopPropagation();
+              setExpanded(!expanded);
+            }}
+            flexShrink={0}
+          >
+            {expanded ? <ChevronUp size="16px" /> : <ChevronDown size="16px" />}
+          </StyledRoundButton>
+        )}
+      </StyledMenuEntry>
+      {expanded &&
+        account?.children?.map(child => (
+          <StyledMenuEntry
+            key={child?.id}
+            href={`/dashboard/${child.slug}`}
+            title={child.name}
+            $isActive={activeSlug === child.slug}
+          >
+            <Flex overflow="hidden" alignItems="center" gridGap="12px">
+              <Avatar ml={4} collective={child} size={32} />
+              <Span truncateOverflow>{child.name}</Span>
+            </Flex>
+          </StyledMenuEntry>
+        ))}
+    </React.Fragment>
+  );
+};
+
+MenuEntry.propTypes = {
+  account: PropTypes.object,
+  activeSlug: PropTypes.string,
+};
+
+const Switcher = () => {
```

Comment:
Not a really clear name, I would suggest. Also for the filename.

```suggestion
const AccountSwitcher = () => {
```

Reply:
Nice, fixed!